$Id: 011a_recognize_glibc_2.3.2_locale_names.diff 1064 2006-01-11 12:26:45Z ender $

Major housecleaning of X11's locale support.

* Add all locales supported by version 2.3.2 of the GNU C Library except
  zh_CN.GB18030, sq_AL.ISO-8859-1, and tg_TJ.KOI8-T.  X has no support for
  the GB18030 and KOI8-T character sets, and glibc appears to disagree with
  most of the world about which character set Albanians use (see, for
  example, iso_8859-2(7)).  This baka gaijin has learned his lesson about
  trying to report l10n bugs in glibc, so I'll let someone else do the
  jousting.  :-P

* Synchronize compose.dir with locale.dir wherever possible (i.e., wherever
  compose tables exist) so that as many supported locales as possible get a
  Compose map.  Comment out references to compose tables that don't
  actually exist.

* Disable support for iu_CA.NUNACOM-8.  X11 doesn't have any support for
  the NUNACOM-8 character set.  (iu_CA.UTF-8 is still supported.)

* Disable support for locales that are neither supported by the GNU C
  Library, nor use standard ISO 639 language or 3166 territory codes.  This
  includes Esperanto (eo_EO), and what I presume is Pennsylvania
  "Dutch" (pd_DE, pd_US).

* Turn off intrusion of Nynorsk into Nyanja's ISO 639 space.  (Nynorsk is
  nn, not ny.)

* Remove bogus "ee" locales for Estonian (et).

* Remove ISO 8859-15 support for languages that don't use that character
  set according to glibc.

* Remove @euro modifier support from locales that don't use that currency
  (i.e., countries not in the Euro zone).

* Ensure we support the @euro modifier for all locales that use it
  (according to glibc), even if the codeset is explicitly specified.  See
  Debian #208137.

* Alias "fi" and "fi_FI" to "fi_FI.ISO8559-1", not "fi_FI.ISO8859-15", for
  compatibility with the GNU C Library.  See Debian #266834.

* Rewrite some comments to be in grammatical English.

This patch by Branden Robinson.
Major "forward port" by Nathanael Nerode.

Partially submitted upstream.  This is so large I don't expect it to all go in at once,
but any bit would help.  --Nathanael

Index: libx11/nls/compose.dir.pre
===================================================================
--- libx11.orig/nls/compose.dir.pre	2013-06-15 13:19:37.000000000 +0200
+++ libx11/nls/compose.dir.pre	2013-06-15 17:34:15.927737867 +0200
@@ -4,8 +4,13 @@
 XCOMM	the second word is full locale name.
 XCOMM
 XCOMM
+XCOMM Note that no Compose tables exist for ISO 8859-11, Microsoft code
+XCOMM pages, Fujitsu's S90 or U90 character sets, or TIS620 yet.
+XCOMM
 iso8859-1/Compose:		C
+iso8859-1/Compose:		aa_DJ.ISO8859-1
 iso8859-1/Compose:		af_ZA.ISO8859-1
+iso8859-15/Compose:		an_ES.ISO8859-15
 iso8859-6/Compose:		ar_AA.ISO8859-6
 iso8859-6/Compose:		ar_AE.ISO8859-6
 iso8859-6/Compose:		ar_BH.ISO8859-6
@@ -70,12 +75,15 @@
 iso8859-1/Compose:		en_BW.ISO8859-1
 iso8859-1/Compose:		en_BZ.ISO8859-1
 iso8859-1/Compose:		en_CA.ISO8859-1
+iso8859-1/Compose:		en_DK.ISO8859-1
+iso8859-15/Compose:		en_DK.ISO8859-15
 iso8859-1/Compose:		en_EN.ISO8859-1
 iso8859-1/Compose:		en_GB.ISO8859-1
 iso8859-15/Compose:		en_GB.ISO8859-15
 iso8859-1/Compose:		en_HK.ISO8859-1
 iso8859-1/Compose:		en_IE.ISO8859-1
 iso8859-15/Compose:		en_IE.ISO8859-15
+iso8859-1/Compose:		en_IN.ISO8859-1
 iso8859-15/Compose:		en_IN.ISO8859-15
 iso8859-1/Compose:		en_JM.ISO8859-1
 iso8859-1/Compose:		en_NZ.ISO8859-1
@@ -124,6 +132,8 @@
 iso8859-15/Compose:		et_EE.ISO8859-15
 iso8859-1/Compose:		eu_ES.ISO8859-1
 iso8859-15/Compose:		eu_ES.ISO8859-15
+iso8859-1/Compose:		eu_FR.ISO8859-1
+iso8859-15/Compose:		eu_FR.ISO8859-15
 isiri-3342/Compose:		fa_IR.ISIRI-3342
 iso8859-1/Compose:		fi_FI.ISO8859-1
 iso8859-15/Compose:		fi_FI.ISO8859-15
@@ -157,11 +167,9 @@
 iso8859-2/Compose:		hu_HU.ISO8859-2
 armscii-8/Compose:		hy_AM.ARMSCII-8
 iso8859-1/Compose:		id_ID.ISO8859-1
-iso8859-15/Compose:		id_ID.ISO8859-15
 iso8859-1/Compose:		is_IS.ISO8859-1
 iso8859-15/Compose:		is_IS.ISO8859-15
 iso8859-1/Compose:		it_CH.ISO8859-1
-iso8859-15/Compose:		it_CH.ISO8859-15
 iso8859-1/Compose:		it_IT.ISO8859-1
 iso8859-15/Compose:		it_IT.ISO8859-15
 ja/Compose:			ja_JP.eucJP
@@ -170,10 +178,10 @@
 georgian-academy/Compose:	ka_GE.GEORGIAN-ACADEMY
 georgian-ps/Compose:		ka_GE.GEORGIAN-PS
 iso8859-1/Compose:		kl_GL.ISO8859-1
-iso8859-15/Compose:		kl_GL.ISO8859-15
 iso8859-1/Compose:		kw_GB.ISO8859-1
 iso8859-14/Compose:		kw_GB.ISO8859-14
 iso8859-15/Compose:		kw_GB.ISO8859-15
+iso8859-10/Compose:		lg_UG.ISO8859-10
 ko/Compose:			ko_KR.eucKR
 ibm-cp1133/Compose:		lo_LA.IBM-CP1133
 mulelao-1/Compose:		lo_LA.MULELAO-1
@@ -202,15 +210,18 @@
 iso8859-15/Compose:		ny_NO.ISO8859-15
 iso8859-1/Compose:		oc_FR.ISO8859-1
 iso8859-15/Compose:		oc_FR.ISO8859-15
-iso8859-1/Compose:		pd_DE.ISO8859-1
-iso8859-15/Compose:		pd_DE.ISO8859-15
-iso8859-1/Compose:		pd_US.ISO8859-1
-iso8859-15/Compose:		pd_US.ISO8859-15
+iso8859-1/Compose:		om_KE.ISO8859-1
+XCOMM Presumably, "pd" is "Pennsylvania 'Dutch'", a U.S. dialect of German.
+XCOMM This is not an ISO language code defined for it, and the GNU C Library
+XCOMM doesn't support it.
+XCOMM iso8859-1/Compose:		pd_DE.ISO8859-1
+XCOMM iso8859-15/Compose:		pd_DE.ISO8859-15
+XCOMM iso8859-1/Compose:		pd_US.ISO8859-1
+XCOMM iso8859-15/Compose:		pd_US.ISO8859-15
 iso8859-1/Compose:		ph_PH.ISO8859-1
 iso8859-2/Compose:		pl_PL.ISO8859-2
 iso8859-1/Compose:		pp_AN.ISO8859-1
 iso8859-1/Compose:		pt_BR.ISO8859-1
-iso8859-15/Compose:		pt_BR.ISO8859-15
 iso8859-1/Compose:		pt_PT.ISO8859-1
 iso8859-15/Compose:		pt_PT.ISO8859-15
 iso8859-2/Compose:		ro_RO.ISO8859-2
@@ -223,6 +234,9 @@
 iso8859-2/Compose:		sh_YU.ISO8859-2
 iso8859-2/Compose:		sk_SK.ISO8859-2
 iso8859-2/Compose:		sl_SI.ISO8859-2
+iso8859-1/Compose:		so_DJ.ISO8859-1
+iso8859-1/Compose:		so_KE.ISO8859-1
+iso8859-1/Compose:		so_SO.ISO8859-1
 iso8859-2/Compose:		sq_AL.ISO8859-2
 iso8859-2/Compose:		sr_CS.ISO8859-2
 iso8859-5/Compose:		sr_CS.ISO8859-5
@@ -261,12 +275,19 @@
 zh_CN.gbk/Compose:		zh_CN.gbk
 zh_CN.gb18030/Compose:		zh_CN.gb18030
 zh_HK.big5/Compose:		zh_HK.big5
+zh_HK.big5hkscs/Compose:	zh_HK.big5-hkscs
+zh_HK.big5hkscs/Compose:	zh_HK.Big5-HKSCS
 zh_HK.big5hkscs/Compose:	zh_HK.big5hkscs
+zh_HK.big5hkscs/Compose:	zh_HK.Big5HKSCS
+zh_CN/Compose:			zh_SG.gb2312
+zh_CN.gbk/Compose:		zh_SG.gbk
 zh_TW.big5/Compose:		zh_TW.big5
 zh_TW/Compose:			zh_TW.eucTW
 iso8859-1/Compose:		zu_ZA.ISO8859-1
 XCOMM
 XCOMM
+en_US.UTF-8/Compose:		aa_ER.UTF-8
+en_US.UTF-8/Compose:		aa_ET.UTF-8
 en_US.UTF-8/Compose:		af_ZA.UTF-8
 am_ET.UTF-8/Compose:		am_ET.UTF-8
 en_US.UTF-8/Compose:		ar_AA.UTF-8
@@ -298,6 +319,7 @@
 en_US.UTF-8/Compose:		bn_BD.UTF-8
 en_US.UTF-8/Compose:		br_FR.UTF-8
 en_US.UTF-8/Compose:		bs_BA.UTF-8
+en_US.UTF-8/Compose:		byn_ER.UTF-8
 en_US.UTF-8/Compose:		ca_AD.UTF-8
 en_US.UTF-8/Compose:		ca_ES.UTF-8
 en_US.UTF-8/Compose:		ca_FR.UTF-8
@@ -358,6 +380,7 @@
 en_US.UTF-8/Compose:		es_VE.UTF-8
 en_US.UTF-8/Compose:		et_EE.UTF-8
 en_US.UTF-8/Compose:		eu_ES.UTF-8
+en_US.UTF-8/Compose:		eu_FR.UTF-8
 en_US.UTF-8/Compose:		fa_IR.UTF-8
 fi_FI.UTF-8/Compose:		fi_FI.UTF-8
 en_US.UTF-8/Compose:		fo_FO.UTF-8
@@ -368,6 +391,8 @@
 en_US.UTF-8/Compose:		fr_LU.UTF-8
 en_US.UTF-8/Compose:		ga_IE.UTF-8
 en_US.UTF-8/Compose:		gd_GB.UTF-8
+en_US.UTF-8/Compose:       gez_ER.UTF-8
+en_US.UTF-8/Compose:       gez_ET.UTF-8
 en_US.UTF-8/Compose:		gl_ES.UTF-8
 en_US.UTF-8/Compose:		gu_IN.UTF-8
 en_US.UTF-8/Compose:		gv_GB.UTF-8
@@ -382,6 +407,7 @@
 en_US.UTF-8/Compose:		it_CH.UTF-8
 en_US.UTF-8/Compose:		it_IT.UTF-8
 en_US.UTF-8/Compose:		iu_CA.UTF-8
+en_US.UTF-8/Compose:		iw_IL.UTF-8
 ja_JP.UTF-8/Compose:		ja_JP.UTF-8
 en_US.UTF-8/Compose:		ka_GE.UTF-8
 en_US.UTF-8/Compose:		kk_KZ.UTF-8
@@ -400,6 +426,7 @@
 en_US.UTF-8/Compose:		mi_NZ.UTF-8
 en_US.UTF-8/Compose:		mk_MK.UTF-8
 en_US.UTF-8/Compose:		ml_IN.UTF-8
+en_US.UTF-8/Compose:		mn_MN.UTF-8
 en_US.UTF-8/Compose:		mr_IN.UTF-8
 en_US.UTF-8/Compose:		ms_MY.UTF-8
 en_US.UTF-8/Compose:		mt_MT.UTF-8
@@ -411,13 +438,19 @@
 en_US.UTF-8/Compose:		nb_NO.UTF-8
 en_US.UTF-8/Compose:		nr_ZA.UTF-8
 en_US.UTF-8/Compose:		nso_ZA.UTF-8
-en_US.UTF-8/Compose:		ny_NO.UTF-8
+XCOMM ny is the language code for Nyanja (spoken in Malawi), not Nynorsk.
+XCOMM See <URL: http://www.ethnologue.com/show_iso639.asp?code=nya >.
+XCOMM en_US.UTF-8/Compose: ny_NO.UTF-8
 en_US.UTF-8/Compose:		oc_FR.UTF-8
 en_US.UTF-8/Compose:		or_IN.UTF-8
+en_US.UTF-8/Compose:       om_ET.UTF-8
 en_US.UTF-8/Compose:		pa_IN.UTF-8
 en_US.UTF-8/Compose:		pa_PK.UTF-8
-en_US.UTF-8/Compose:		pd_DE.UTF-8
-en_US.UTF-8/Compose:		pd_US.UTF-8
+XCOMM Presumably, "pd" is "Pennsylvania 'Dutch'", a U.S. dialect of German.
+XCOMM This is not an ISO language code defined for it, and the GNU C Library
+XCOMM doesn't support it.
+XCOMM en_US.UTF-8/Compose: pd_DE.UTF-8
+XCOMM en_US.UTF-8/Compose: pd_US.UTF-8
 en_US.UTF-8/Compose:		ph_PH.UTF-8
 en_US.UTF-8/Compose:		pl_PL.UTF-8
 en_US.UTF-8/Compose:		pp_AN.UTF-8
@@ -433,9 +466,11 @@
 en_US.UTF-8/Compose:		se_NO.UTF-8
 en_US.UTF-8/Compose:		sh_BA.UTF-8
 en_US.UTF-8/Compose:		sh_YU.UTF-8
+en_US.UTF-8/Compose:		sid_ET.UTF-8
 en_US.UTF-8/Compose:		si_LK.UTF-8
 en_US.UTF-8/Compose:		sk_SK.UTF-8
 en_US.UTF-8/Compose:		sl_SI.UTF-8
+en_US.UTF-8/Compose:		so_ET.UTF-8
 en_US.UTF-8/Compose:		sq_AL.UTF-8
 sr_CS.UTF-8/Compose:		sr_CS.UTF-8
 en_US.UTF-8/Compose:		sr_ME.UTF-8
@@ -451,6 +486,7 @@
 th_TH.UTF-8/Compose:		th_TH.UTF-8
 en_US.UTF-8/Compose:		ti_ER.UTF-8
 en_US.UTF-8/Compose:		ti_ET.UTF-8
+en_US.UTF-8/Compose:		tig_ER.UTF-8
 en_US.UTF-8/Compose:		tl_PH.UTF-8
 en_US.UTF-8/Compose:		tn_ZA.UTF-8
 en_US.UTF-8/Compose:		tr_TR.UTF-8
Index: libx11/nls/locale.alias.pre
===================================================================
--- libx11.orig/nls/locale.alias.pre	2013-06-15 13:19:37.307909585 +0200
+++ libx11/nls/locale.alias.pre	2013-06-15 17:14:09.031751432 +0200
@@ -311,6 +311,12 @@
 en_CA.ISO-8859-1:				en_CA.ISO8859-1
 en_CA.ISO_8859-1:				en_CA.ISO8859-1
 en_CA.utf8:					en_CA.UTF-8
+en_DK:						en_DK.ISO8859-1
+en_DK.iso88591:					en_DK.ISO8859-1
+en_DK.ISO-8859-1:				en_DK.ISO8859-1
+en_DK.iso885915:				en_DK.ISO8859-15
+en_DK.ISO-8859-15:				en_DK.ISO8859-15
+en_DK.utf8:					en_DK.UTF-8
 en_DL.utf8:					en_DL.UTF-8
 en_GB:						en_GB.ISO8859-1
 en_GB.88591:					en_GB.ISO8859-1
@@ -364,6 +370,16 @@
 en_US.ISO8859-15@euro:				en_US.ISO8859-15
 en_US.utf8:					en_US.UTF-8
 EN_US.UTF-8:					en_US.UTF-8
+en_US.88591.en:				en_US.ISO8859-1
+en_US.iso88591:				en_US.ISO8859-1
+en_US.ISO-8859-1:				en_US.ISO8859-1
+en_US.ISO_8859-1:				en_US.ISO8859-1
+en_US.iso885915:				en_US.ISO8859-15
+en_US.ISO-8859-15:				en_US.ISO8859-15
+en_US.8859-15:					en_US.ISO8859-15
+en_US.ISO8859-15@euro:				en_US.ISO8859-15
+en_US.utf8:					en_US.UTF-8
+EN_US.UTF-8:					en_US.UTF-8
 en_ZA:						en_ZA.ISO8859-1
 en_ZA.88591:					en_ZA.ISO8859-1
 en_ZA.88591.en:					en_ZA.ISO8859-1
@@ -932,7 +948,6 @@
 pt_BR.iso88591:					pt_BR.ISO8859-1
 pt_BR.ISO-8859-1:				pt_BR.ISO8859-1
 pt_BR.ISO_8859-1:				pt_BR.ISO8859-1
-pt_BR.iso885915:				pt_BR.ISO8859-15
 pt_BR.utf8:					pt_BR.UTF-8
 pt_PT:						pt_PT.ISO8859-1
 pt_PT.88591:					pt_PT.ISO8859-1
@@ -993,11 +1008,11 @@
 XCOMM These lines should be deleted in the future.
 sh:						sr_RS.UTF-8@latin
 sh_BA.ISO8859-2@bosnia:				sr_CS.ISO8859-2
-sh_HR.iso88592:					hr_HR.ISO8859-2
 sh_YU:						sr_RS.UTF-8@latin
 sh_YU.utf8:					sr_RS.UTF-8@latin
 si:						si_LK.UTF-8
 si_LK:						si_LK.UTF-8
+sid_ET:                        sid_ET.UTF-8
 sk:						sk_SK.ISO8859-2
 sk_SK:						sk_SK.ISO8859-2
 sk_SK.iso88592:					sk_SK.ISO8859-2
@@ -1210,7 +1225,7 @@
 american.iso88591:				en_US.ISO8859-1
 arabic.iso88596:				ar_AA.ISO8859-6
 bokmal:						nb_NO.ISO8859-1
-bokmål:						nb_NO.ISO8859-1
+bokmï¿½:						nb_NO.ISO8859-1
 bulgarian:					bg_BG.CP1251
 c-french.iso88591:				fr_CA.ISO8859-1
 catalan:					ca_ES.ISO8859-1
@@ -1225,11 +1240,11 @@
 dutch:						nl_NL.ISO8859-1
 dutch.iso88591:					nl_BE.ISO8859-1
 eesti:						et_EE.ISO8859-1
-english.iso88591:				en_EN.ISO8859-1
+english.iso88591:				en_US.ISO8859-1
 estonian:					et_EE.ISO8859-1
 finnish:					fi_FI.ISO8859-1
 finnish.iso88591:				fi_FI.ISO8859-1
-français:					fr_FR.ISO8859-1
+franï¿½is:					fr_FR.ISO8859-1
 french:						fr_FR.ISO8859-1
 french.iso88591:				fr_CH.ISO8859-1
 galego:						gl_ES.ISO8859-1
Index: libx11/nls/locale.dir.pre
===================================================================
--- libx11.orig/nls/locale.dir.pre	2013-06-15 13:19:37.307909585 +0200
+++ libx11/nls/locale.dir.pre	2013-06-15 17:29:34.107741034 +0200
@@ -6,8 +6,11 @@
 XCOMM
 
 C/XLC_LOCALE:				C
+iso8859-1/XLC_LOCALE:			aa_DJ.ISO8859-1
 iso8859-1/XLC_LOCALE:			af_ZA.ISO8859-1
+iso8859-15/XLC_LOCALE:			an_ES.ISO8859-15
 iso8859-6/XLC_LOCALE:			ar_AA.ISO8859-6
+iso8859-6/XLC_LOCALE:			ar_AE.ISO8859-6
 iso8859-6/XLC_LOCALE:			ar_BH.ISO8859-6
 iso8859-6/XLC_LOCALE:			ar_DZ.ISO8859-6
 iso8859-6/XLC_LOCALE:			ar_EG.ISO8859-6
@@ -36,6 +39,7 @@
 iso8859-1/XLC_LOCALE:			br_FR.ISO8859-1
 iso8859-14/XLC_LOCALE:			br_FR.ISO8859-14
 iso8859-15/XLC_LOCALE:			br_FR.ISO8859-15
+iso8859-2/XLC_LOCALE:			bs_BA.ISO8859-2
 iso8859-1/XLC_LOCALE:			ca_AD.ISO8859-1
 iso8859-15/XLC_LOCALE:			ca_AD.ISO8859-15
 iso8859-1/XLC_LOCALE:			ca_ES.ISO8859-1
@@ -59,7 +63,7 @@
 iso8859-1/XLC_LOCALE:			de_DE.ISO8859-1
 iso8859-15/XLC_LOCALE:			de_DE.ISO8859-15
 iso8859-1/XLC_LOCALE:			de_LI.ISO8859-1
-iso8859-1/XLC_LOCALE:			de_LI.ISO8859-15
+iso8859-15/XLC_LOCALE:			de_LI.ISO8859-15
 iso8859-1/XLC_LOCALE:			de_LU.ISO8859-1
 iso8859-15/XLC_LOCALE:			de_LU.ISO8859-15
 iso8859-7/XLC_LOCALE:			el_GR.ISO8859-7
@@ -71,12 +75,15 @@
 iso8859-1/XLC_LOCALE:			en_BZ.ISO8859-1
 iso8859-1/XLC_LOCALE:			en_BW.ISO8859-1
 iso8859-1/XLC_LOCALE:			en_CA.ISO8859-1
+iso8859-1/XLC_LOCALE:			en_DK.ISO8859-1
+iso8859-15/XLC_LOCALE:			en_DK.ISO8859-15
 iso8859-1/XLC_LOCALE:			en_EN.ISO8859-1
 iso8859-1/XLC_LOCALE:			en_GB.ISO8859-1
 iso8859-15/XLC_LOCALE:			en_GB.ISO8859-15
 iso8859-1/XLC_LOCALE:			en_HK.ISO8859-1
 iso8859-1/XLC_LOCALE:			en_IE.ISO8859-1
 iso8859-15/XLC_LOCALE:			en_IE.ISO8859-15
+iso8859-1/XLC_LOCALE:			en_IN.ISO8859-1
 iso8859-15/XLC_LOCALE:			en_IN.ISO8859-15
 iso8859-1/XLC_LOCALE:			en_JM.ISO8859-1
 iso8859-1/XLC_LOCALE:			en_NZ.ISO8859-1
@@ -90,7 +97,8 @@
 iso8859-1/XLC_LOCALE:			en_ZA.ISO8859-1
 iso8859-15/XLC_LOCALE:			en_ZA.ISO8859-15
 iso8859-1/XLC_LOCALE:			en_ZW.ISO8859-1
-iso8859-3/XLC_LOCALE:			eo_EO.ISO8859-3
+XCOMM The EO territory doesn't exist in ISO 3166.
+XCOMM iso8859-3/XLC_LOCALE:			eo_EO.ISO8859-3
 iso8859-3/XLC_LOCALE:			eo_XX.ISO8859-3
 iso8859-1/XLC_LOCALE:			es_AR.ISO8859-1
 iso8859-1/XLC_LOCALE:			es_BO.ISO8859-1
@@ -125,6 +133,8 @@
 iso8859-15/XLC_LOCALE:			et_EE.ISO8859-15
 iso8859-1/XLC_LOCALE:			eu_ES.ISO8859-1
 iso8859-15/XLC_LOCALE:			eu_ES.ISO8859-15
+iso8859-1/XLC_LOCALE:			eu_FR.ISO8859-1
+iso8859-15/XLC_LOCALE:			eu_FR.ISO8859-15
 isiri-3342/XLC_LOCALE:			fa_IR.ISIRI-3342
 iso8859-1/XLC_LOCALE:			fi_FI.ISO8859-1
 iso8859-15/XLC_LOCALE:			fi_FI.ISO8859-15
@@ -176,6 +186,7 @@
 iso8859-1/XLC_LOCALE:			kw_GB.ISO8859-1
 iso8859-14/XLC_LOCALE:			kw_GB.ISO8859-14
 iso8859-15/XLC_LOCALE:			kw_GB.ISO8859-15
+iso8859-10/XLC_LOCALE:			lg_UG.ISO8859-10
 ibm-cp1133/XLC_LOCALE:			lo_LA.IBM-CP1133
 mulelao-1/XLC_LOCALE:			lo_LA.MULELAO-1
 iso8859-4/XLC_LOCALE:			lt_LT.ISO8859-4
@@ -203,8 +214,12 @@
 iso8859-1/XLC_LOCALE:			ny_NO.ISO8859-15
 iso8859-1/XLC_LOCALE:			oc_FR.ISO8859-1
 iso8859-15/XLC_LOCALE:			oc_FR.ISO8859-15
-iso8859-1/XLC_LOCALE:			pd_DE.ISO8859-1
-iso8859-15/XLC_LOCALE:			pd_DE.ISO8859-15
+iso8859-1/XLC_LOCALE:			om_KE.ISO8859-1
+XCOMM Presumably, "pd" is "Pennsylvania 'Dutch'", a U.S. dialect of German.
+XCOMM This is not an ISO language code defined for it, and the GNU C Library
+XCOMM doesn't support it.
+XCOMM iso8859-1/XLC_LOCALE:			pd_DE.ISO8859-1
+XCOMM iso8859-15/XLC_LOCALE:			pd_DE.ISO8859-15
 iso8859-1/XLC_LOCALE:			pd_US.ISO8859-1
 iso8859-15/XLC_LOCALE:			pd_US.ISO8859-15
 iso8859-1/XLC_LOCALE:			ph_PH.ISO8859-1
@@ -224,6 +239,9 @@
 iso8859-2/XLC_LOCALE: 			sh_YU.ISO8859-2
 iso8859-2/XLC_LOCALE: 			sk_SK.ISO8859-2
 iso8859-2/XLC_LOCALE: 			sl_SI.ISO8859-2
+iso8859-1/XLC_LOCALE:			so_DJ.ISO8859-1
+iso8859-1/XLC_LOCALE:			so_KE.ISO8859-1
+iso8859-1/XLC_LOCALE:			so_SO.ISO8859-1
 iso8859-2/XLC_LOCALE: 			sq_AL.ISO8859-2
 iso8859-2/XLC_LOCALE:			sr_CS.ISO8859-2
 iso8859-5/XLC_LOCALE:			sr_CS.ISO8859-5
@@ -238,8 +256,8 @@
 iso8859-15/XLC_LOCALE:			sv_SE.ISO8859-15
 tscii-0/XLC_LOCALE:			ta_IN.TSCII-0
 koi8-c/XLC_LOCALE:			tg_TJ.KOI8-C
-th_TH/XLC_LOCALE:			th_TH.TIS620
 iso8859-11/XLC_LOCALE:			th_TH.ISO8859-11
+th_TH/XLC_LOCALE:			th_TH.TIS620
 iso8859-1/XLC_LOCALE: 			tl_PH.ISO8859-1
 iso8859-15/XLC_LOCALE:			tn_ZA.ISO8859-15
 iso8859-9/XLC_LOCALE:			tr_TR.ISO8859-9
@@ -263,10 +281,14 @@
 zh_CN.gb18030/XLC_LOCALE:		zh_CN.gb18030
 zh_HK.big5/XLC_LOCALE:			zh_HK.big5
 zh_HK.big5hkscs/XLC_LOCALE:		zh_HK.big5hkscs
+zh_CN/XLC_LOCALE:			zh_SG.gb2312
+zh_CN.gbk/XLC_LOCALE			zh_SG.gbk
 zh_TW.big5/XLC_LOCALE:			zh_TW.big5
 zh_TW/XLC_LOCALE:			zh_TW.eucTW
 iso8859-1/XLC_LOCALE:			zu_ZA.ISO8859-1
 XCOMM Note: The UTF-8 locales don't work correctly yet. Work in progress.
+en_US.UTF-8/XLC_LOCALE:			aa_ER.UTF-8
+en_US.UTF-8/XLC_LOCALE:			aa_ET.UTF-8
 en_US.UTF-8/XLC_LOCALE:			af_ZA.UTF-8
 en_US.UTF-8/XLC_LOCALE:			am_ET.UTF-8
 en_US.UTF-8/XLC_LOCALE:			ar_AA.UTF-8
@@ -297,6 +319,7 @@
 en_US.UTF-8/XLC_LOCALE:			bn_IN.UTF-8
 en_US.UTF-8/XLC_LOCALE:                 bo_IN.UTF-8
 en_US.UTF-8/XLC_LOCALE:			br_FR.UTF-8
+en_US.UTF-8/XLC_LOCALE:			byn_ER.UTF-8
 en_US.UTF-8/XLC_LOCALE:			ca_AD.UTF-8
 en_US.UTF-8/XLC_LOCALE:			ca_ES.UTF-8
 en_US.UTF-8/XLC_LOCALE:			ca_FR.UTF-8
@@ -334,7 +357,8 @@
 en_US.UTF-8/XLC_LOCALE:			en_US.UTF-8
 en_US.UTF-8/XLC_LOCALE:			en_ZA.UTF-8
 en_US.UTF-8/XLC_LOCALE:			en_ZW.UTF-8
-en_US.UTF-8/XLC_LOCALE:			eo_EO.UTF-8
+XCOMM The EO territory doesn't exist in ISO 3166.
+XCOMM en_US.UTF-8/XLC_LOCALE:			eo_EO.UTF-8
 en_US.UTF-8/XLC_LOCALE:			eo_XX.UTF-8
 en_US.UTF-8/XLC_LOCALE:			es_AR.UTF-8
 en_US.UTF-8/XLC_LOCALE:			es_BO.UTF-8
@@ -358,6 +382,7 @@
 en_US.UTF-8/XLC_LOCALE:			es_VE.UTF-8
 en_US.UTF-8/XLC_LOCALE:			et_EE.UTF-8
 en_US.UTF-8/XLC_LOCALE:			eu_ES.UTF-8
+en_US.UTF-8/XLC_LOCALE:			eu_FR.UTF-8
 en_US.UTF-8/XLC_LOCALE:			fa_IR.UTF-8
 en_US.UTF-8/XLC_LOCALE:			fi_FI.UTF-8
 en_US.UTF-8/XLC_LOCALE:			fo_FO.UTF-8
@@ -368,6 +393,8 @@
 en_US.UTF-8/XLC_LOCALE:			fr_LU.UTF-8
 en_US.UTF-8/XLC_LOCALE:			ga_IE.UTF-8
 en_US.UTF-8/XLC_LOCALE:			gd_GB.UTF-8
+en_US.UTF-8/XLC_LOCALE:            gez_ER.UTF-8
+en_US.UTF-8/XLC_LOCALE:            gez_ET.UTF-8
 en_US.UTF-8/XLC_LOCALE:			gl_ES.UTF-8
 en_US.UTF-8/XLC_LOCALE:			gu_IN.UTF-8
 en_US.UTF-8/XLC_LOCALE:			gv_GB.UTF-8
@@ -382,6 +409,7 @@
 en_US.UTF-8/XLC_LOCALE:			it_CH.UTF-8
 en_US.UTF-8/XLC_LOCALE:			it_IT.UTF-8
 en_US.UTF-8/XLC_LOCALE:			iu_CA.UTF-8
+en_US.UTF-8/XLC_LOCALE:			iw_IL.UTF-8
 ja_JP.UTF-8/XLC_LOCALE:			ja_JP.UTF-8
 en_US.UTF-8/XLC_LOCALE:			ka_GE.UTF-8
 en_US.UTF-8/XLC_LOCALE:			kk_KZ.UTF-8
@@ -400,9 +428,11 @@
 en_US.UTF-8/XLC_LOCALE:			mi_NZ.UTF-8
 en_US.UTF-8/XLC_LOCALE:			mk_MK.UTF-8
 en_US.UTF-8/XLC_LOCALE:			ml_IN.UTF-8
+en_US.UTF-8/XLC_LOCALE:            mn_MN.UTF-8
 en_US.UTF-8/XLC_LOCALE:			mr_IN.UTF-8
 en_US.UTF-8/XLC_LOCALE:			ms_MY.UTF-8
 en_US.UTF-8/XLC_LOCALE:			mt_MT.UTF-8
+en_US.UTF-8/XLC_LOCALE:            ne_NP.UTF-8
 en_US.UTF-8/XLC_LOCALE:			nl_BE.UTF-8
 en_US.UTF-8/XLC_LOCALE:			nl_NL.UTF-8
 en_US.UTF-8/XLC_LOCALE:			nb_NO.UTF-8
@@ -410,13 +440,19 @@
 en_US.UTF-8/XLC_LOCALE:			no_NO.UTF-8
 en_US.UTF-8/XLC_LOCALE:			nr_ZA.UTF-8
 en_US.UTF-8/XLC_LOCALE:			nso_ZA.UTF-8
-en_US.UTF-8/XLC_LOCALE:			ny_NO.UTF-8
+XCOMM ny is the language code for Nyanja (spoken in Malawi), not Nynorsk.
+XCOMM See <URL: http://www.ethnologue.com/show_iso639.asp?code=nya >.
+XCOMM en_US.UTF-8/XLC_LOCALE:          ny_NO.UTF-8
 en_US.UTF-8/XLC_LOCALE:			oc_FR.UTF-8
+en_US.UTF-8/XLC_LOCALE:            om_ET.UTF-8
 en_US.UTF-8/XLC_LOCALE:			or_IN.UTF-8
 en_US.UTF-8/XLC_LOCALE:			pa_IN.UTF-8
 en_US.UTF-8/XLC_LOCALE:			pa_PK.UTF-8
-en_US.UTF-8/XLC_LOCALE:			pd_DE.UTF-8
-en_US.UTF-8/XLC_LOCALE:			pd_US.UTF-8
+XCOMM Presumably, "pd" is "Pennsylvania 'Dutch'", a U.S. dialect of German.
+XCOMM This is not an ISO language code defined for it, and the GNU C Library
+XCOMM doesn't support it.
+XCOMM en_US.UTF-8/XLC_LOCALE:          pd_DE.UTF-8
+XCOMM en_US.UTF-8/XLC_LOCALE:          pd_US.UTF-8
 en_US.UTF-8/XLC_LOCALE:			ph_PH.UTF-8
 en_US.UTF-8/XLC_LOCALE:			pl_PL.UTF-8
 en_US.UTF-8/XLC_LOCALE:			pp_AN.UTF-8
@@ -430,11 +466,13 @@
 en_US.UTF-8/XLC_LOCALE:                 sd_IN.UTF-8
 en_US.UTF-8/XLC_LOCALE:                 sd_IN@devanagari.UTF-8
 en_US.UTF-8/XLC_LOCALE:			se_NO.UTF-8
+en_US.UTF-8/XLC_LOCALE:            sid_ET.UTF-8
 en_US.UTF-8/XLC_LOCALE:			sh_BA.UTF-8
 en_US.UTF-8/XLC_LOCALE:			sh_YU.UTF-8
 en_US.UTF-8/XLC_LOCALE:			si_LK.UTF-8
 en_US.UTF-8/XLC_LOCALE:			sk_SK.UTF-8
 en_US.UTF-8/XLC_LOCALE:			sl_SI.UTF-8
+en_US.UTF-8/XLC_LOCALE:            so_ET.UTF-8
 en_US.UTF-8/XLC_LOCALE:			sq_AL.UTF-8
 en_US.UTF-8/XLC_LOCALE:			sr_CS.UTF-8
 en_US.UTF-8/XLC_LOCALE:			sr_ME.UTF-8
@@ -450,6 +488,7 @@
 th_TH.UTF-8/XLC_LOCALE:			th_TH.UTF-8
 en_US.UTF-8/XLC_LOCALE:			ti_ER.UTF-8
 en_US.UTF-8/XLC_LOCALE:			ti_ET.UTF-8
+en_US.UTF-8/XLC_LOCALE:            tig_ER.UTF-8
 en_US.UTF-8/XLC_LOCALE:			tl_PH.UTF-8
 en_US.UTF-8/XLC_LOCALE:			tn_ZA.UTF-8
 en_US.UTF-8/XLC_LOCALE:			tr_TR.UTF-8
